docs/spark-writes.md

```diff
@@ -339,74 +331,55 @@ USING iceberg
-SELECT id, data, category, ts FROM another_table
-ORDER BY iceberg_bucket16(id)
-```
-
-If you're inserting data with DataFrame, you can use the function like below:
-
-```scala
-data.sortWithinPartitions(expr("iceberg_bucket16(id)"))
-    .writeTo("prod.db.sample")
-    .append()
-```
 
+There are 3 options for `write.distribution-mode`
+
+* `none` - This is the previous default for Iceberg.<p> This mode does not require any shuffles or sort to be performed
+automatically by Spark. Because no work is done automatically by Spark, the data must be either locally or globally 
+sorted manually by partition value. To reduce the number of files produced during writing, using a global sort is recommended.<p> 
+A local sort can be avoided by using the Spark [write fanout](#write-properties) property but this will cause all file handles to 
+remain open until each write task has completed. 
+* `hash` - This mode is the new default and requests that Spark uses a hash-based exchange to shuffle the incoming
+write data before writing. Practically, this means that each row is hashed based on the row's partition value and then placed
+in a corresponding Spark task based upon that value. Further division and coalescing of tasks may take place based on 
+the [Spark's Adaptive Query planning](#controlling-file-sizes).
+* `range` - This mode requests that Spark perform a range based exchanged to shuffle the data before writing. This is
+a two stage procedure which is more expensive than the `hash` mode. The first stage samples the data to be written based
+on the partition and sort columns, this information is then used in the second stage to shuffle data into tasks. Each
+task gets an exclusive range of the input data which clusters the data by partition and also globally sorts it.
+While this is more expensive than the hash distribution, the global ordering can be beneficial for read performance if
+sorted columns are used during queries. Further division and coalescing of tasks may take place based on
+  the [Spark's Adaptive Query planning](#controlling-file-sizes).
+
+
+## Controlling File Sizes
+
+When writing data to Iceberg with Spark, it's important to note that Spark cannot write a file larger than a Spark 
+task. This means although Iceberg will always roll over a file when it grows to 
+[`write.target-file-size-bytes`](../configuration/#write-properties), a file
+will not be able to grow to that size if the task is not large enough. The
+on disk file size will also be much smaller than the Spark task size since the on disk data will be both compressed 
+and in columnar format as opposed to Spark's uncompressed row representation. This means a 100 megabyte task will 
+always corrospond to on an on disk file of much less than 100 megabytes even when writing to a single Iceberg partition.
+
+To control what data ends up in each task the user must either use a [`write distribution mode`](#writing-distribution-modes) 
+or manually repartition the data. 
+
```

Comment:
Do we need `<p>` here and above or not?  As we had some above

Reply:
Here it's an actual paragraph, above it's to put an endline in a bullet list

Reply:
For future readers I was wrong, the correct thing to do is to end the line with two spaces